### PR TITLE
Graphs for Grade Data under tabs

### PR DESCRIPTION
**Pull request by thef1stofhername**
I have created graphs in the course gradable status page. There are visualizations of the total scores for each student in histogram form, the autograder scores in histogram form, and a bar chart of average scores for each component compared to the full points possible for each. This is open for extension and modification easily at any time. The graphs are pulled from plotly, an open source project.
updoot_06142019Orosz

### CODE REVIEW

File: site/app/libraries/database/txt.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+SELECT * FROM(
```
Comment (zjxiaohan) [inline site/app/libraries/database/txt.sql]: The content of this `txt.sql` file seems to be the same as the code in `DatabaseQueries.php`. Is this file needed or used somewhere?

File: site/app/templates/grading/electronic/Status.twig [hunk 2]
```diff
@@ -1,164 +1,32 @@
+<head>
+    <!-- Plotly.js -->
+    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
```
Comment (zjxiaohan) [inline site/app/templates/grading/electronic/Status.twig]: Could we add this JS file locally through `addInternalJS` instead of fetching it from elsewhere?

File: site/app/templates/grading/electronic/Status.twig [hunk 3]
```diff
@@ -1,164 +1,32 @@
...
+    {#<script type="text/javascript" language="javascript" src="{{ base_url }}vendor/jquery/jquery.min.js"></script>#}
```
Comment (zjxiaohan) [inline site/app/templates/grading/electronic/Status.twig]: This comment can be deleted.

File: site/app/libraries/database/txt.sql (hunk 1 above)
Comment (thef1stofhername) [reply site/app/libraries/database/txt.sql]: Mistakenly added from test I was doing. Removed

File: site/app/templates/grading/electronic/Status.twig [hunk 4]
```diff
@@ -176,13 +39,568 @@
...
+        // Get all elements with class="tablinks" and remove the class "active"
+        tablinks = document.getElementsByClassName("tablinks");
+        for (i = 0; i < tablinks.length; i++) {
+            tablinks[i].className = tablinks[i].className.replace(" active", "");
+        }
+
+        // Show the current tab, and add an "active" class to the button that opened the tab
+        document.getElementById(statName).style.display = "block";
+        //to make active, do this, not good for current implementation
+        //evt.currentTarget.className += " active";
```
Comment (zjxiaohan) [inline site/app/templates/grading/electronic/Status.twig]: I do feel there needs to be some sort of mechanism to tell me which page I am in though... What do you think?

File: site/app/libraries/database/DatabaseQueries.php [hunk 5]
```diff
@@ -957,6 +957,57 @@ public function getAverageAutogradedScores($g_id, $section_key, $is_team) {
...
+                ON gd.g_id=auto.g_id AND gd_{$user_or_team_id}=auto.{$user_or_team_id}
+                INNER JOIN {$users_or_teams} AS {$u_or_t} ON {$u_or_t}.{$user_or_team_id} = auto.{$user_or_team_id}
+                WHERE gc.g_id=? AND {$u_or_t}.{$section_key} IS NOT NULL
+              )AS parts_of_comp
+            )AS comp
+            GROUP BY gd_id, autograding
+          )g 
+      ", array($g_id));
+        if(count($this->course_db->rows()) == 0){
+          echo("why");
```
Comment (zjxiaohan) [inline site/app/libraries/database/DatabaseQueries.php]: The updated code works well in my local machine and it looks good. One suggestion: How about replacing `why` with something more description if `why` indicates an error or an exception or something.

Comment (thef1stofhername) [reply site/app/libraries/database/DatabaseQueries.php]: Nope that was just a sanity check I thought I took out... but I looked in the wrong place (line 922)

File: site/app/templates/grading/electronic/Status.twig (hunk 4 above)
Comment (thef1stofhername) [reply site/app/templates/grading/electronic/Status.twig]: fixedtabs 2019-06-18 10-36

File: site/app/templates/grading/electronic/Status.twig [hunk 6]
```diff
@@ -590,15 +590,15 @@
         }
 
         // Get all elements with class="tablinks" and remove the class "active"
```
Comment (zjxiaohan) [inline site/app/templates/grading/electronic/Status.twig]: The comment is outdated.

File: site/app/templates/grading/electronic/Status.twig [hunk 7]
```diff
@@ -590,15 +590,15 @@
...
-        tablinks = document.getElementsByClassName("tablinks");
+        tablinks = document.getElementsByClassName("btn btn-primary");
         for (i = 0; i < tablinks.length; i++) {
             tablinks[i].className = tablinks[i].className.replace(" active", "");
         }
+        console.log(tablinks);
```
Comment (zjxiaohan) [inline site/app/templates/grading/electronic/Status.twig]: Buttons work nicely now! Please remove the `console.log` and I believe it is good to merge. :)
